### PR TITLE
Add option to allow special-use subfolders

### PR DESCRIPTION
**Pull request by MichaelMenge**
implement #2995 for master

### CODE REVIEW

File: imap/mboxlist.c [hunk 1]
```diff
@@ -1974,8 +1974,8 @@ static int _rename_check_specialuse(const char *oldname, const char *newname)
         strarray_t *check = strarray_split(protect, NULL, STRARRAY_TRIM);
         strarray_t *uses = strarray_split(buf_cstring(&attrib), NULL, 0);
         if (strarray_intersect_case(uses, check)) {
-            /* then target must be a single-depth mailbox too */
-            if (strarray_size(mbname_boxes(new)) != 1)
+            /* then if llowspecialusesubfolders is not enabled the target must be a single-depth mailbox too */
```
Comment (elliefm) [inline imap/mboxlist.c:1977]: Typo:

```suggestion
            /* then if allowspecialusesubfolders is not enabled the target must be a single-depth mailbox too */
```

File: lib/imapoptions [hunk 2]
```diff
@@ -172,6 +172,14 @@ are listed with ``<none>''.
 /* Defaults to enabled.  If disabled, disallows the use of the SETACL
    command at all via IMAP. */
 
+{ "allowspecialusesubfolder", 0, SWITCH, "3.0.14" }
```
Comment (elliefm) [inline lib/imapoptions:175]: Don't hardcode version numbers here, just put "UNRELEASED" and the real version will be put in when a release is made.

```suggestion
{ "allowspecialusesubfolder", 0, SWITCH, "UNRELEASED" }
```

**Review by elliefm — CHANGES_REQUESTED** [review]
Sorry this took so long to get to.  Looks good, couple of small details that need fixing though.

**Comment by MichaelMenge** [comment]
@elliefm these changes are fine with me, can you add them or do i need to update the pull request?

**Review by elliefm — APPROVED** [review]
Thanks!

**Pull request merged**